src/hd.rs

```diff
@@ -117,23 +127,20 @@ impl KeyTree {
         })
     }
 
-    /// Recover a [KeyTree] for its mnemonic seed phrase.
-    ///
-    /// Input a mnemonic phrase, in the form of bytes, i.e.,
-    /// "digital-apollo-aroma--rival-artist-rebel" initializing a key tree.
+    /// Recover a [KeyTree] from its mnemonic seed phrase.
     ///
-    /// We require that the input mnemonic phrase is 12 words (equivalently, 16 bytes of seed), and
-    /// return a decoding error if the mnemonic is not decoded successfully or if the input length
-    /// is incorrect.
-    pub fn from_mnemonic(mnemonic: &[u8]) -> Result<Self, argon2::Error> {
-        let mut decoded_seed = Vec::<u8>::new();
-        decode(mnemonic, &mut decoded_seed).map_err(|_| argon2::Error::DecodingFail)?;
-        if decoded_seed.len() != SEED_LENGTH {
-            return Err(argon2::Error::DecodingFail);
+    /// See [BIP-39](https://github.com/bitcoin/bips/blob/master/bip-0039.mediawiki) for more
+    /// details on deterministically recovering entropy from a mnemonic.
+    pub fn from_mnemonic(mnemonic: &Mnemonic) -> Self {
+        Self {
```

Comment:
BIP39 seems to allow for multiple length of seeds/mnemonic. Do we want to enforce 12 words like we did earlier?

Reply:
I don't see a strong reason to enforce it. It has a minimum of 12 words so we don't have to be too worried about security. We will only generate 12-word mnemonics, which is all the UI cares about. If someone wants to generate a longer mnemonic for extra entropy, that seems fine. WDYT?

Reply:
My worries was
- if there is an UI that takes, for instance, 24 words, and passes the first 12 words to generate the seed
- someone else use a CLI to generate the seed with the same 24 words
those two will not match. 

It is good that our UI takes care of it.